x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/settings/RelevanceSettingsService.java

```diff
@@ -160,6 +168,16 @@ private static XContentBuilder getInternalIndexMapping() {
                 .startObject("_meta")
                 .field("version", Version.CURRENT)
                 .endObject()
+                .startArray("dynamic_templates")
```

Comment:
This is because I don't want to explicitly define boosts. Mappings would get pretty complicated otherwise.

Reply:
We're already seeing pretty complicated mappings - do they have to be built in code or is it possible to pull from `/src/main/resources`? Maybe a theoretical question for @ioanatia

Reply:
Yes, it would be so much easier if we can parse them from json. I'm not sure how to do that.

Reply:
Not sure about parsing the mapping from JSON, but we can take an approach like:

https://github.com/elastic/elasticsearch/blob/73cdc7b80ab42c5335512d86eb09b1ac1a085f85/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java#L138-L538

Reply:
Thanks @ioanatia - if we have to use the builders that may be a bit more readable. 

If we're using builders to create rather complex json, we can also consider breaking out those builders into their own mini classes.

Reply:
We're already using the same builder, just a different (chained) syntax. I rewrote this piece using Ioana's linked example. I think it's more readable this way. Thanks @ioanatia .